failed to upload new image with CDI [port-forward]

Hi, I need your help. is my first time with CDI
I'm trying to upload a new image (qcow2) from my local machine.
I tried a lot of things:
- I created  dv & pv & pvc and then try run folowing commad - ./virtctl image-upload --pvc-name=minio-pv-claim   --namespace default  --size=5Gi --image-path /home/tacohen_nfs/Centos74-clean.qcow2  --uploadproxy-url=https://127.0.0.1:18443 --insecure

- I created only dv and try to run follwing command - 
./virtctl image-upload dv centos74  --namespace default  --size=5Gi --image-path /home/tacohen_nfs/Centos74-clean.qcow2  --uploadproxy-url=https://127.0.0.1:18443 --insecure
'''
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  name: upload-datavolume
spec:
  source:
      upload: {}
  pvc:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 500Mi
'''

can't figure out what's the simple way to upload a new disk image from my local machine.
please help.



Events:
  Type     Reason            Age        From               Message
  ----     ------            ----       ----               -------
  Warning  FailedScheduling  <unknown>  default-scheduler  persistentvolumeclaim "upload-datavolume-scratch" not found
  Warning  FailedScheduling  <unknown>  default-scheduler  running "VolumeBinding" filter plugin for pod "cdi-upload-upload-datavolume": pod has unbound immediate PersistentVolumeClaims
  Warning  FailedScheduling  <unknown>  default-scheduler  running "VolumeBinding" filter plugin for pod "cdi-upload-upload-datavolume": pod has unbound immediate PersistentVolumeClaims

if I create manually the PV + PV scratch and the PVC 
I get the following error 



/virtctl image-upload dv centos74  --namespace default   --size=5Gi --image-path /home/tacohen_nfs/Centos74-clean.qcow2  --uploadproxy-url=https://10.111.12.16:31001  --insecure  --no-create
panic: assignment to entry in nil map

goroutine 1 [running]:
kubevirt.io/kubevirt/pkg/virtctl/imageupload.ensurePVCSupportsUpload(0x7f0b25b34118, 0xc00038a500, 0xc0000bafc0, 0x7f0b25b34118, 0xc00038a500, 0x8)
        pkg/virtctl/imageupload/imageupload.go:506 +0xb0
kubevirt.io/kubevirt/pkg/virtctl/imageupload.(*command).run(0xc000245d40, 0xc0003e8a00, 0xc0003ee3c0, 0x2, 0xa, 0x0, 0x0)
        pkg/virtctl/imageupload/imageupload.go:234 +0xda4
kubevirt.io/kubevirt/pkg/virtctl/imageupload.NewImageUploadCommand.func1(0xc0003e8a00, 0xc0003ee3c0, 0x2, 0xa, 0x0, 0x0)
        pkg/virtctl/imageupload/imageupload.go:123 +0x6d
kubevirt.io/kubevirt/vendor/github.com/spf13/cobra.(*Command).execute(0xc0003e8a00, 0xc0003ee280, 0xa, 0xa, 0xc0003e8a00, 0xc0003ee280)
        vendor/github.com/spf13/cobra/command.go:826 +0x465
kubevirt.io/kubevirt/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003ce000, 0x0, 0xc00014a3d0, 0x0)
        vendor/github.com/spf13/cobra/command.go:914 +0x2fc
kubevirt.io/kubevirt/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        vendor/github.com/spf13/cobra/command.go:864
kubevirt.io/kubevirt/pkg/virtctl.Execute()
        pkg/virtctl/root.go:101 +0x48
main.main()
        cmd/virtctl/virtctl.go:31 +0x22

--- a/pkg/virtctl/imageupload/imageupload.go
+++ b/pkg/virtctl/imageupload/imageupload.go
@@ -503,6 +503,9 @@ func ensurePVCSupportsUpload(client kubernetes.Interface, pvc *v1.PersistentVolu
 	_, hasAnnotation := pvc.Annotations[uploadRequestAnnotation]
 
 	if !hasAnnotation {
+		if pvc.GetAnnotations() == nil {
+			pvc.SetAnnotations(make(map[string]string, 0))
+		}
 		pvc.Annotations[uploadRequestAnnotation] = ""
 		pvc, err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(pvc)
 		if err != nil {

--- a/pkg/virtctl/imageupload/imageupload_test.go
+++ b/pkg/virtctl/imageupload/imageupload_test.go
@@ -97,6 +97,26 @@ var _ = Describe("ImageUpload", func() {
 		return pvc
 	}
 
+	pvcSpecNoAnnotationMap := func() *v1.PersistentVolumeClaim {
+		quantity, _ := resource.ParseQuantity(pvcSize)
+
+		pvc := &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dvName,
+				Namespace: "default",
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: quantity,
+					},
+				},
+			},
+		}
+
+		return pvc
+	}
+
 	pvcSpecWithUploadAnnotation := func() *v1.PersistentVolumeClaim {
 		spec := pvcSpec()
 		spec.Annotations = map[string]string{
@@ -356,6 +376,7 @@ var _ = Describe("ImageUpload", func() {
 		},
 			Entry("PVC with upload annotation", pvcSpecWithUploadAnnotation()),
 			Entry("PVC without upload annotation", pvcSpec()),
+			Entry("PVC without upload annotation and no annotation map", pvcSpecNoAnnotationMap()),
 		)
 
 		It("PVC exists deprecated args", func() {

